sound/soc/sof/core.c

```diff
@@ -492,6 +491,9 @@ int snd_sof_device_remove(struct device *dev)
 	if (!IS_ERR_OR_NULL(pdata->pdev_mach))
 		platform_device_unregister(pdata->pdev_mach);
 
+	/* remove DSP */
+	snd_sof_remove(sdev);
+
```

Comment:
Good catch, but still one question unclear to me:

In your commit message it is stated "snd_hdac_ext_bus_device_remove() to remove the codec which unregisters the component driver thereby also removing the topology.", per my understanding, this only remove the codec soc component(not the soc component we register here?), how does it removing the topology.

Reply:
@keyonjie take a look at __snd_soc_unregister_component() in soc-core.c. It removes the tplg for all components who's dev matches the input argument "dev". Now, in our case both the hdmic codec component device and the SOF component device are the same and thats why snd_hdac_ext_bus_device_remove() results in removing the topology

Reply:
yes, if we share the "dev" in both hdmi codec and SOF component, that explains, thanks.